### PR TITLE
(TECH-66) Offer List

### PR DESCRIPTION
**Pull request by letbueno**
# Description

- Add page offer list

# Necessary changes

- Routes to offer list
- Page show all offers
- Tests

## Does this pull request close any open issues?

- Put here the issue that must be closed with this PR
- eg: closes [#TECH-66](https://ribon.atlassian.net/jira/software/c/projects/TECH/boards/26?modal=detail&selectedIssue=TECH-66)

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Dangerous changes
- [ ] Chore (change that does not effect how the code works, eg: change color)
- [ ] Tests

## How to test

- Go to offers page and verify if list all offers

### CODE REVIEW

File: src/pages/OffersPage/OffersItems/index.tsx [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+  searchTerm: string;
+};
+
+function OffersItems({ offers, searchTerm }: Props) {
+  const { green30, red30 } = theme.colors;
+  const { t } = useTranslation("translation", {
+    keyPrefix: "offersPage.offersListSection.offersItems",
+  });
+
+  function filterOffers(nonFilteredCauses: any) {
```
Comment (NickNish09) [inline src/pages/OffersPage/OffersItems/index.tsx:23]: I think that here would be nice to type this nonFilteredCauses, what do u think

File: src/pages/OffersPage/OffersItems/index.tsx [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+    return nonFilteredCauses.filter((offerData: any) => {
```
Comment (NickNish09) [inline src/pages/OffersPage/OffersItems/index.tsx:24]: ditto

File: src/pages/OffersPage/OffersItems/index.tsx [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+      } else {
+        return null;
+      }
+    });
+  }
+
+  function renderOffers() {
+    return (
+      offers &&
+      filterOffers(offers).map((offer: any) => (
```
Comment (NickNish09) [inline src/pages/OffersPage/OffersItems/index.tsx:40]: ditto

File: src/pages/OffersPage/OffersListSection/index.tsx [hunk 4]
```diff
@@ -0,0 +1,84 @@
...
+  const { t } = useTranslation("translation", {
+    keyPrefix: "offersPage.offersListSection",
+  });
+  const [currentOffers, setCurrentOffers] = useState<Offer[]>([]);
+  const [pageCount, setPageCount] = useState(0);
+  const [itemOffset, setItemOffset] = useState(0);
+  const itemsPerPage = 10;
+  const [searchTerm, setSearchTerm] = useState("");
+
+  const fetchCauses = useCallback(async () => {
```
Comment (NickNish09) [inline src/pages/OffersPage/OffersListSection/index.tsx:21]: isnt this method fetching offers instead?

Comment (letbueno) [reply src/pages/OffersPage/OffersListSection/index.tsx]: yes kkkkk

**Review by NickNish09 — APPROVED** [review]

**Pull request merged**